CC/Sounder/config.cc

```diff
@@ -126,18 +118,48 @@ Config::Config(const std::string& jsonfile, const std::string& directory,
     cal_tx_gain_.shrink_to_fit();
 
     sample_cal_en_ = tddConf.value("sample_calibrate", false);
+    cal_node_ = tddConf.value("calib_node", "");
     imbalance_cal_en_ = tddConf.value("imbalance_calibrate", false);
     beam_sweep_ = tddConf.value("beamsweep", false);
     beacon_ant_ = tddConf.value("beacon_antenna", 0);
     max_frame_ = tddConf.value("max_frame", 0);
 
+    // Load serials file (loads hub, sdr, and rrh serials)
+    std::string serials_str;
+    auto serials_file_ = tddConf.value("serials_file", "./files/topology.json");
+    Utils::loadTDDConfig(serials_file_, serials_str);
+    const auto j_serials = json::parse(serials_str, nullptr, true, true);
+    json serials_conf;
+
+    num_cells_ = tddConf.value("cells", 1);
     MLPD_TRACE("Number cells: %zu\n", num_cells_);
+    hub_ids_.resize(num_cells_);
     bs_sdr_ids_.resize(num_cells_);
     n_bs_sdrs_.resize(num_cells_);
     n_bs_antennas_.resize(num_cells_);
     num_bs_sdrs_all_ = 0;
-    for (size_t i = 0u; i < num_cells_; i++) {
-        Utils::loadDevices(bs_sdr_file_.at(i), bs_sdr_ids_.at(i));
+
+    reciprocal_calib_ = tddConf.value("internal_measurement", false);
+    for (size_t i = 0; i < num_cells_; i++) {
+        std::string cell_str = "Cell" + std::to_string(i);
+        ss << j_serials[0].value(cell_str, serials_conf);
+        serials_conf = json::parse(ss);
+        ss.str(std::string());
+        ss.clear();
+
+        hub_ids_.at(i) = serials_conf.value("hub", "");
+        auto sdr_serials = serials_conf.value("sdr", json::array());
+        bs_sdr_ids_.at(i).assign(sdr_serials.begin(), sdr_serials.end());
+
+        // Remove cal node if not doing reciprocal calibration
+        if (!reciprocal_calib_) {
```

Comment:
Instead of this, we can modify pyfaros so it can separate the calib node from the rest of the sdrs. A calib node is connected to chain 7, so it should be easy to identify it.

Reply:
Can we assume that's always the case? (including POWDER). For example, one of the BS already deployed lists chains 1, 2, 2, 3, 4, 4, 5, 5. what happens when we install the calib node, wouldn't it be listed as chain 6?

Reply:
For POWDER, I think it should be 6. We can also have condition that if the length is 1, then consider it as a reference node. For RENEW, the reference node is always on chain 7.